File: Did We Get Everything Covered?.cpp
#include<bits/stdc++.h>
using namespace std;

void solve(){
    int n,k,m;
    cin>>n>>k>>m;
    string s;
    cin>>s;
    set<char>st;
    string ans="";
    for(int i=0;i<m;i++){
        st.insert(s[i]);
        if(st.size()==k){
            ans+=s[i];
            st.clear();
        }
    }
    if(ans.length()>=n){
        // more than n blocks are present;
        cout<<"YES"<<'\n';
        return;
    }
    else{
        cout<<"NO"<<'\n';
        char miss;
        for(int i=0;i<k;i++){
            char ch=(char)('a'+i);
            if(st.find(ch)==st.end()){
                miss=ch;
            }
        }
        while(ans.length()<n){
            ans+=miss;
        }
        cout<<ans<<'\n';
        return;
    }
}

int main(){
    int t;cin>>t;
    while(t--) solve();
}

File: Did We Get Everything Covered?.py
def solve():
    n, k, m = map(int, input().split())
    s = input()
    st = set()
    ans = ""
    for i in range(m):
        st.add(s[i])
        if len(st) == k:
            ans += s[i]
            st.clear()
    if len(ans) >= n:
        print("YES")
        return
    else:
        print("NO")
        miss = None
        for i in range(k):
            ch = chr(ord('a') + i)
            if ch not in st:
                miss = ch
        while len(ans) < n:
            ans += miss
        print(ans)
        return

t = int(input())
for _ in range(t):
    solve()

